### PR TITLE
[codex] Fix CosDPR encoder loading with newer Transformers

### PR DESCRIPTION
**Pull request by lintool**
## Summary

- Load the cosDPR projection head explicitly under Transformers 5.x so checkpoint weights are used instead of random initialization.
- Put cosDPR encoders in eval mode and encode under `torch.no_grad()`.
- Update the cosDPR encoder test expectations to match the checkpoint-loaded vector.

## Validation

- `python -m unittest tests.base.encoder.test_encoder_model_cosdpr`
- `python -m py_compile pyserini/encode/_cosdpr.py tests/base/encoder/test_encoder_model_cosdpr.py`
- `python -m pyserini.search.faiss --threads 16 --batch-size 512 --index msmarco-v1-passage.cosdpr-distil --topics tmp/q2.tsv --encoder castorini/cosdpr-distil --output tmp/run.q2.cosdpr.fixed.txt --hits 1000`

Additional debug note: qid `2` retrieves judged doc `4339068` at rank 1 after the fix. A partial full-dev run showed healthy completed-query `R@1000 = 0.9796`, but was interrupted before producing all `6,980,000` lines.

### CODE REVIEW

**Comment by lintool** [comment]
@sahel-sh @lilyjge latest whack-a-mole in Transformers 5 upgrade and fixing all 2CRs.

**Comment by lintool** [comment]
This works now:

```
% python -m pyserini.2cr.msmarco --collection v1-passage --condition cosdpr-distil.faiss-flat.pytorch --directory runs/ --display-commands

# Running condition "cosdpr-distil.faiss-flat.pytorch": cosDPR-distil: Faiss flat, PyTorch

  - topic_key: msmarco-v1-passage.dev

python -m pyserini.search.faiss \
  --threads 16 --batch-size 512 \
  --index msmarco-v1-passage.cosdpr-distil \
  --topics msmarco-v1-passage.dev \
  --encoder castorini/cosdpr-distil \
  --output runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dev.txt

Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 201/201 [00:00<00:00, 23209.95it/s]
Attempting to initialize prebuilt index msmarco-v1-passage.cosdpr-distil.
/Users/jimmylin/.cache/pyserini/indexes/faiss-flat.msmarco-v1-passage.cosdpr-distil.20221023.83565019175c79fcc5f8d99fb1bd43ca already exists, skipping download.
Initializing msmarco-v1-passage.cosdpr-distil...
Running msmarco-v1-passage.dev topics, saving to runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dev.txt...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6980/6980 [1:03:40<00:00,  1.83it/s]

python -m pyserini.eval.trec_eval -c -M 10 -m recip_rank msmarco-v1-passage.dev runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dev.txt

    MRR@10 : 0.3896 [OK]

python -m pyserini.eval.trec_eval -c -m recall.1000 msmarco-v1-passage.dev runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dev.txt

    R@1K   : 0.9796 [OK]

  - topic_key: dl19-passage

python -m pyserini.search.faiss \
  --threads 16 --batch-size 512 \
  --index msmarco-v1-passage.cosdpr-distil \
  --topics dl19-passage \
  --encoder castorini/cosdpr-distil \
  --output runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl19.txt

Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 201/201 [00:00<00:00, 55909.22it/s]
Attempting to initialize prebuilt index msmarco-v1-passage.cosdpr-distil.
/Users/jimmylin/.cache/pyserini/indexes/faiss-flat.msmarco-v1-passage.cosdpr-distil.20221023.83565019175c79fcc5f8d99fb1bd43ca already exists, skipping download.
Initializing msmarco-v1-passage.cosdpr-distil...
Running dl19-passage topics, saving to runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl19.txt...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 43/43 [04:03<00:00,  5.65s/it]

python -m pyserini.eval.trec_eval -c -l 2 -m map dl19-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl19.txt

    MAP    : 0.4656 [OK]

python -m pyserini.eval.trec_eval -c -m ndcg_cut.10 dl19-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl19.txt

    nDCG@10: 0.7250 [OK]

python -m pyserini.eval.trec_eval -c -l 2 -m recall.1000 dl19-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl19.txt

    R@1K   : 0.8201 [OK]

  - topic_key: dl20-passage

python -m pyserini.search.faiss \
  --threads 16 --batch-size 512 \
  --index msmarco-v1-passage.cosdpr-distil \
  --topics dl20-passage \
  --encoder castorini/cosdpr-distil \
  --output runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl20.txt

Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 201/201 [00:00<00:00, 59504.17it/s]
Attempting to initialize prebuilt index msmarco-v1-passage.cosdpr-distil.
/Users/jimmylin/.cache/pyserini/indexes/faiss-flat.msmarco-v1-passage.cosdpr-distil.20221023.83565019175c79fcc5f8d99fb1bd43ca already exists, skipping download.
Initializing msmarco-v1-passage.cosdpr-distil...
Running dl20-passage topics, saving to runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl20.txt...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [03:49<00:00,  1.15s/it]

python -m pyserini.eval.trec_eval -c -l 2 -m map dl20-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl20.txt

    MAP    : 0.4876 [OK]

python -m pyserini.eval.trec_eval -c -m ndcg_cut.10 dl20-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl20.txt

    nDCG@10: 0.7025 [OK]

python -m pyserini.eval.trec_eval -c -l 2 -m recall.1000 dl20-passage runs/run.msmarco-v1-passage.cosdpr-distil.faiss-flat.pytorch.dl20.txt

    R@1K   : 0.8533 [OK]

                                                                          TREC 2019                TREC 2020            MS MARCO dev
                                                                   MAP    nDCG@10    R@1K    MAP    nDCG@10    R@1K    MRR@10    R@1K
                                                                   ----------------------    ----------------------    --------------
cosDPR-distil: Faiss flat, PyTorch                                 0.4656  0.7250  0.8201    0.4876  0.7025  0.8533    0.3896  0.9796


Start time: 2026-05-22 12:30:10
End time: 2026-05-22 13:44:19
Total elapsed time: 4449s ~1.2hr
```

**Review by sahel-sh — APPROVED** [review]
